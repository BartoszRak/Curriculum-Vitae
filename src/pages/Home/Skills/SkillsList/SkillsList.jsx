import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { Typography } from '@material-ui/core'

import useStyle from './SkillsList.style'

export function SkillsList({ classes: overridingClasses, className, data }) {
  const classes = { ...useStyle(), ...overridingClasses }
  return (
    <div className={classNames(classes.root, className)}>
      {data.map(skill => (
        <Typography className={classNames(classes.skill, classes.item)} color="secondary" key={skill.name} variant="h5">
          {skill.name}
        </Typography>
      ))}
      <Typography aria-hidden className={classNames(classes.sum, classes.item)} color="secondary" variant="h5">
        {data.length}
      </Typography>
    </div>
  )
}

SkillsList.propTypes = {
  classes: PropTypes.object,
  className: PropTypes.string,
  data: PropTypes.arrayOf(PropTypes.shape({
    advancement: PropTypes.number,
    name: PropTypes.string,
  })),
}

export default SkillsList
