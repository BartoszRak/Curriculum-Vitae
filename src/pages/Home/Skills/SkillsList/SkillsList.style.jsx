export default ({ palette, spacing }) => ({
  item: {
    marginBottom: spacing.unit * 1,
    marginRight: spacing.unit * 2,
    padding: `${spacing.unit * 0.5}px ${spacing.unit * 2.5}px`,
  },
  root: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  skill: {
    background: palette.secondary.main,
    borderRadius: spacing.unit * 2,
    color: palette.secondary.contrastText,
    fontWeight: 'normal',
  },
  sum: {
    border: `2px dashed ${palette.secondary.main}`,
    borderRadius: spacing.unit * 2,
    color: palette.secondary.main,
    fontWeight: 'bold',
  },
})
