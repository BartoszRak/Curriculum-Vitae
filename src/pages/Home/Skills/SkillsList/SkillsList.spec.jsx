import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { SkillsList } from './SkillsList'

jest.unmock('@material-ui/core')

let data

describe('component: SkillsList', () => {
  afterEach(cleanup)

  beforeEach(() => {
    data = [{
      advancement: 1,
      name: '1 Test skill name',
    }, {
      advancement: 3,
      name: '2 Test skill name',
    }]
  })

  describe('rendering', () => {
    test('match snapshot ', () => {
      const wrapper = render(<SkillsList data={data} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render skills', () => {
      const { getByText } = render(<SkillsList data={data} />)

      data.forEach(skill => {
        const skillEl = getByText(skill.name)
        expect(skillEl).toBeVisible()
      })
    })

    test('render valid summary', () => {
      const { getByTestId } = render(<SkillsList data={data} />)
      const skillSummaryEl = getByTestId('skills-list-summary')

      expect(skillSummaryEl).toBeVisible()
    })
  })
})
