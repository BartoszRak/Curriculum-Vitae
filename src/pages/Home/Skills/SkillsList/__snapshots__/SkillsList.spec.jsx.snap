// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: SkillsList rendering match snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-2"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary makeStyles-skill-3 makeStyles-item-1"
        >
          1 Test skill name
        </h5>
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary makeStyles-skill-3 makeStyles-item-1"
        >
          2 Test skill name
        </h5>
        <h5
          aria-hidden="true"
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary makeStyles-sum-4 makeStyles-item-1"
          data-testid="skills-list-summary"
        >
          2
        </h5>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-2"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary makeStyles-skill-3 makeStyles-item-1"
      >
        1 Test skill name
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary makeStyles-skill-3 makeStyles-item-1"
      >
        2 Test skill name
      </h5>
      <h5
        aria-hidden="true"
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary makeStyles-sum-4 makeStyles-item-1"
        data-testid="skills-list-summary"
      >
        2
      </h5>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
